pyo3-build-config/src/impl_.rs

```diff
@@ -677,6 +668,39 @@ pub fn cross_compiling(
     }))
 }
 
+fn is_not_cross_compiling(
+    host: &str,
+    target_arch: &str,
+    target_vendor: &str,
+    target_os: &str,
+) -> bool {
+    let target_triple = format!(
+        "{}-{}-{}",
+        if target_arch == "x86" {
+            "i686"
+        } else {
+            target_arch
+        },
+        target_vendor,
+        if target_os == "macos" {
+            "darwin"
+        } else {
+            target_os
+        }
+    );
+
+    // Not cross-compiling if arch-vendor-os is all the same
+    // e.g. x86_64-unknown-linux-musl on x86_64-unknown-linux-gnu host
+    //      x86_64-pc-windows-gnu on x86_64-pc-windows-msvc host
+    host.starts_with(&target_triple)
+        // Not cross-compiling to compile for 32-bit Python from windows 64-bit
+        || (target_triple == "i686-pc-windows" && host.starts_with("x86_64-pc-windows"))
```

Comment:
The failure of the Windows job suggests that something is still off here, but debugging it still hindered by the `ensure!` formatting not working.

Reply:
So this should be
```rust
target_triple == "i686-windows-pc"
```
i.,e. the other of `windows` and `pc` being switched?

Reply:
Actually no, the issue seems to be with the call site switching os and vendor.